### ISSUE TITLE
Half precision support

### ISSUE THREAD

**Issue by xooxit** [spcl/gemm_hls#18]
Hello

I wanna build the project with 'half precision' datatype, so i try 

`"cmake ../ -DMM_DATA_TYPE=half -DMM_PARALLELISM_N=32 -DMM_PARALLELISM_M=8 -DMM_MEMORY_TILE_SIZE_N=512 -DMM_MEMORY_TILE_SIZE_M=512"`

But it fails during synthesis (`"make synthesis"` on command line) with the 

```
INFO: [HLS 200-10] Analyzing design file '/home/lab/yong/gemm_hls/kernel/Compute.cpp' ...
ERROR: [HLS 214-124] use of undeclared identifier 'Data_t': /home/lab/yong/gemm_hls/build/Config.h:34
ERROR: [HLS 214-124] use of undeclared identifier 'Data_t': /home/lab/yong/gemm_hls/build/Config.h:35
```

How can I synthesize with half-precision?
btw, very nice design, learned a lot from it.

**Comment by definelicht** [comment]
Hi xooxit! I can reproduce this locally, but I'm not sure yet why this is happening. I will look into this and let you know!

**Comment by definelicht** [comment]
There was a flag missing to synthesis, this has been fixed with 832d4a4.

Thank you for reporting this!
